students/charlie_england/lesson02/assignment/src/charges_calc.py

```diff
@@ -0,0 +1,138 @@
+'''
+Returns total price paid for individual rentals
+'''
+import argparse
+import json
+import datetime
+import math
+import logging
+
+def parse_cmd_arguments():
+    """
+    creates an argument parser, adds 3 arguments
+    i for input, o for output and optional d for debugging with 3 levels
+    """
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-d', '--debug', help='level of debug, 0 is default = not set, 1 = critical, 2 = warning, 3 = debug', required=False)
+
+    return parser.parse_args()
+
+
+def load_rentals_file(filename):
+    """
+    Opens JSON file and returns data,
+    logs a critical error if there is an error
+    """
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+            #Json decode error, expecting property name enclosed in double quotes
+        except Exception as exception:
+            logging.critical(f'Unable to load json file with name {filename} due to {exception}')
+    return data
+
+def calculate_additional_fields(data):
+    """
+    Takes data and iterates through the keys/values
+    series of try, except blocks that will log an error if something occurs
+    converts rental start/end to datetime
+    if the total days is negative: it will switch the start/end days and recalculate total days logging this only if level 3 (debug logging level)
+    
+    Generally the switch in the rental start/end would be a warning level debug for me, 
+    for this exercise I made it a debug level due to there being so many of them
+    and it clogging up the log. I would write a program to fix the data in the source.json if I was to continue
+    """
+    for key, value in data.items():
+        try:
+            rental_start = datetime.datetime.strptime(value['rental_start'], '%m/%d/%y')
+            rental_end = datetime.datetime.strptime(value['rental_end'], '%m/%d/%y')
+        except Exception as exception:
+            logging.warning(f'FAILED - rental start/rental end {key} with exception {exception}')
+        try:
+            value['total_days'] = (rental_end - rental_start).days
+            if value['total_days'] < 0:
+                logging.debug(f'total days is negative, switching rental start and end for {key}')
+                foo = rental_start
```

Comment:
Instead of using multiple values to switch the values around, you could do something much simpler.

i. Take the absolute value to always get a positive number of days
ii. value['total_days'] = (rental_start - rental_end).days

I think the first option is neater